app/services/sign_in/user_loader.rb

```diff
@@ -20,45 +20,48 @@ def find_user
 
     def reload_user
       validate_account_and_session
-      user_identity.uuid = user_uuid
-      current_user.uuid = user_uuid
+      user_identity.uuid = access_token.user_uuid
+      current_user.uuid = access_token.user_uuid
       current_user.last_signed_in = session.created_at
       current_user.save && user_identity.save
       current_user
     end
 
     def validate_account_and_session
-      raise Errors::UserAccountNotFoundError, 'Invalid User UUID' unless user_account
       raise Errors::SessionNotFoundError, 'Invalid Session Handle' unless session
     end
 
     def user_attributes
       {
-        mhv_icn: user_account.icn,
-        idme_uuid: retrieve_credential_uuid(:idme_uuid),
-        logingov_uuid: retrieve_credential_uuid(:logingov_uuid),
-        loa: { current: loa, highest: LOA::THREE }
+        mhv_icn: session.user_account.icn,
+        idme_uuid: user_verification.idme_uuid,
+        logingov_uuid: user_verification.logingov_uuid,
+        loa: loa,
+        email: session.credential_email,
+        authn_context: user_verification.credential_type,
+        sign_in: sign_in
       }
     end
 
     def loa
-      user_account.icn ? LOA::THREE : LOA::ONE
+      current_loa = session.user_account.verified? ? LOA::THREE : LOA::ONE
+      { current: current_loa, highest: LOA::THREE }
     end
 
-    def user_uuid
-      @user_uuid ||= access_token.user_uuid
+    def sign_in
+      { service_name: credential_service_name, auth_broker: SignIn::Constants::Auth::BROKER_CODE }
     end
 
-    def session
-      @session ||= OAuthSession.find_by(handle: access_token.session_handle)
+    def credential_service_name
+      user_verification.credential_type == 'logingov' ? 'logingov' : 'idme'
```

Comment:
It was my understanding of the `sign_in[:service_name]` that it is supposed to represent the CSP a user authenticates with, don't we need to allow this to set for DSLogon and MHV as well?

Reply:
Yeah, good point. I updated this and fixed the services that were using `service_name` differently as well